test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -482,20 +482,53 @@ def test_module_device_mismatches_device_id(self):
             )
 
     @skip_if_lt_x_gpu(2)
-    def test_multi_device_not_supported(self):
-        """Tests that wrapping a multi-device module (i.e. with submodules on
-        both GPU and CPU) with FSDP raises an error."""
+    def test_cpu_gpu_module(self):
+        """Tests a CPU + GPU module raises warning if device_id is passed
+        in, errors if device_id is not.
+        """
+        torch.cuda.set_device(self.rank)
+
 
-        class MultiDeviceModule(nn.Module):
+        class CPUGPUModule(nn.Module):
             def __init__(self):
                 super().__init__()
                 self.a = nn.Linear(1, 1).cuda()
                 self.b = nn.Linear(1, 1)
 
+
+        with self.assertWarnsRegex(
+            UserWarning, "FSDP wrapping module with params"
+        ):
+            FSDP(CPUGPUModule(), device_id=torch.cuda.current_device())
+
+        # without device_id, we hit an error
+        with self.assertRaisesRegex(
+            RuntimeError, "please pass in device_id"
+        ):
+            FSDP(CPUGPUModule())
+
+    @skip_if_lt_x_gpu(2)
+    def test_multigpu_module(self):
+        """
+        Module on multiple GPUs wrapped in FSDP should raise an error.
+        """
+        torch.cuda.set_device(self.rank)
```

Comment:
curious: Do we not already set this in the test setup? (I feel like we rely on it being set.)